code/modules/food_and_drinks/machinery/microwave.dm

```diff
@@ -466,20 +477,36 @@
 	. = ..()
 	if((machine_stat & NOPOWER) && operating)
 		pre_fail()
-		eject()
+		eject(force = TRUE) // monkestation edit: microwave "enhancements"
 
 /obj/machinery/microwave/proc/loop_finish(mob/cooker)
 	operating = FALSE
 
 	var/metal_amount = 0
+	var/shouldnt_open = FALSE // monkestation edit: microwave "enhancements"
+	var/dont_eject = FALSE // monkestation edit: microwave "enhancements"
 	for(var/obj/item/cooked_item in ingredients)
+		// var/sigreturn = cooked_item.microwave_act(src, cooker, randomize_pixel_offset = ingredients.len)
+		// if(sigreturn & COMPONENT_MICROWAVE_SUCCESS)
+		// 	if(isstack(cooked_item))
+		// 		var/obj/item/stack/cooked_stack = cooked_item
+		// 		dirty += cooked_stack.amount
+		// 	else
+		// 		dirty++
```

Comment:
?

Reply:
there's too many edits between the lines that I'd rather comment out the original, and paste an edit below it.